### ISSUE TITLE
Loss scaling is incorrect when using gradient_accumulation_steps > 1

### ISSUE THREAD

**Issue by BitPhinix** [linkedin/Liger-Kernel#802]
### 🐛 Describe the bug

Fixed it for this one specific case here, sadly don't have the time to put in a proper pr for all models for all cases: https://github.com/BitPhinix/Liger-Kernel/commit/c455526c4707e622d4e7518255b14e018e43228d

TLDR:

Ever since https://github.com/huggingface/transformers/pull/34191

Transformers expect all models that take in `kwargs` to scale the loss by `num_items_in_batch`.

Right now, the loss is effectively multiplied by the number of gradient accumulation steps steps when using liger kernels

### Versions

all

**Comment by Tcc0403** [comment]
AFAIK, most hugginface models with gradient accumulation fix are calling `self.loss_function(logits, labels, self.vocab_size, **kwargs)` which is [`fixed_cross_entropy()`](https://github.com/huggingface/transformers/blob/4b47b2b8ea9252e36669397243b770505f953d69/src/transformers/loss/loss_utils.py#L45) under the hood. Liger has a similar function defined as [LigerForCausalLMLoss()](https://github.com/linkedin/Liger-Kernel/blob/02bac056330eb06711ef5dbb0325cc2087702a3d/src/liger_kernel/transformers/model/loss_utils.py#L33).

Any model that is originally calling `self.loss_function(logits, labels, self.vocab_size, **kwargs)` should have `LigerForCausalLMLoss()` patched already. I might overlook so feel free to tell me if there wasn't the case.

Regarding gemma3 multimodal or some specific models, in huggingface implementation, there's also a PR about it recently. https://github.com/huggingface/transformers/pull/37208

In gemma3 multidomal [forward()](https://github.com/huggingface/transformers/blob/896e9cea1ade521b2648f4798218550f6c72190c/src/transformers/models/gemma3/modeling_gemma3.py#L1123), it doesn't take `num_items_in_batch` and scale the loss either. That's why I didn't add loss scaling to it as it would differ from huggface's training result.
